plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2075,8 +2047,17 @@ protected StartAnswer execute(StartCommand cmd) {
             //
             // Setup ROOT/DATA disk devices
             //
+            if (Arrays.stream(sortedDisks).filter(disk -> disk.getType() == Volume.Type.ISO).count() > 1 && deployAsIs) {
+                sortedDisks = Arrays.stream(sortedDisks).filter(vol -> ((vol.getPath() != null &&
```

Comment:
Looks good but could also be extracted to a method and be tested with unit tests to increase understanding